DesignSafe-CI/portal
# WP-710: For apps with no inputs, only show launch button

## Overview: ##
For certain apps like OpenSees-Interactive, the only input is maxMinutes. Provide a simple job launch option for such apps.
See design in [WP-710](https://tacc-main.atlassian.net/browse/WP-710) 

## PR Status: ##


## Related Jira tickets: ##

* [WP-710](https://tacc-main.atlassian.net/browse/WP-710)

## Summary of Changes: ##
1. Add App form notes option to hide max minutes. See [Apps PR 58](https://github.com/TACC/WMA-Tapis-Templates/pull/58). 
2. In DesignSafe Portal code, add notes optional fields for hideMaxMinutes and an launch text.
3. Hide Max Minutes when the notes flag indicates so.
4. When the steps have no fields (or we have no steps), provide a quick launch UI. The launch button varies between interactive and other apps.
5. jobLaunchDescription is unused in app, but could be useful if users/PI wants to add custom text.

## Testing Steps: ##
1. Open https://designsafe.dev/rw/workspace/opensees-interactive?appVersion=3.7.0 and see if you can launch the app with the launch button session. The UI matches what is shown in [WP-710](https://tacc-main.atlassian.net/browse/WP-710) 

2. Regression testing: https://designsafe.dev/rw/workspace/openfoam. In this app, the regular form should show. Also, in the configuration step, the field dependency works as normal (change from small to large to queues, varies maxMinutes and other fields).

3. Verified the json we submit from server logs:
```
op=submitJob info={'body': {'operation': 'submitJob', 'licenseType': None, 'isInteractive': True, 'job': {'archiveOnAppError': True, 'appId': 'opensees-interactive', 'appVersion': '3.7.0', 'execSystemId': 'wma-exec-01', 'fileInputs': [], 'parameterSet': {'envVariables': [{'key': '_UserProjects', 'value': ''}]}, 'name': 'opensees-interactive-3.7.0_2024-10-08T19:22:21', 'archiveSystemId': 'designsafe.storage.default', 'archiveSystemDir': 'foouser/tapis-jobs-archive/${JobCreateDate}/${JobName}-${JobUUID}'}}}
```


## UI Photos:
<img width="2137" alt="Screenshot 2024-10-08 at 2 21 51 PM" src="https://github.com/user-attachments/assets/c952a0ee-ceaa-4f94-8407-8a9adb58b471">

## Notes: ##
